lib/logflare_web/live/alerts/alerts_live.ex

```diff
@@ -61,19 +61,31 @@ defmodule LogflareWeb.AlertsLive do
     alert =
       if alert_id do
         Alerting.get_alert_query_by(id: alert_id, user_id: socket.assigns.user_id)
-        |> Alerting.preload_alert_query()
+        |> case do
+          nil -> nil
+          alert -> Alerting.preload_alert_query(alert)
+        end
       end
 
     socket = assign(socket, :alert, alert)
 
     socket =
-      if socket.assigns.live_action == :edit do
+      if socket.assigns.live_action == :edit and alert != nil do
         assign(socket, :changeset, Alerting.change_alert_query(alert))
       else
         assign(socket, :changeset, nil)
       end
 
-    {:noreply, socket}
+    if !!alert_id and !alert do
```

Comment:
You can make it a bit more explicit, which may be more readable than the negations:

```suggestion
    if alert_id != nil and alert == nil do
```